app/services/analytics/ecf_school_service.rb

```diff
@@ -41,6 +44,8 @@ def update_school_analytics
                 delivery_partner: school.active_partnerships.first&.delivery_partner&.name || school.partnerships.first&.delivery_partner&.name,
                 chosen_cip: school.school_cohorts.find { |sc| sc.cohort_id == current_cohort_id }&.core_induction_programme&.name,
                 active_participants: school.school_cohorts.find { |sc| sc.cohort_id == current_cohort_id }&.active_ecf_participants&.any? || false,
+                sparsity: school.local_authority_district&.district_sparsities&.find { |ds| ds.start_year <= current_cohort_year && ds.end_year > current_cohort_year }&.any? || false,
+                pupil_premium: school.pupil_premiums.find { |pp| pp.start_year == current_cohort_year }&.uplift?,
```

Comment:
```suggestion
                sparsity: school.sparsity_uplift?(current_cohort_year),
                pupil_premium: school.pupil_premium_uplift?(current_cohort_year),
```

I'm pretty sure these are doing the same thing.

Reply:
yeah that was my first try. It's n+1 hell though

Reply:
Yeah, that would take us from 50 queries to 50k queries